@shared/api/internal.ts

```diff
@@ -310,6 +315,92 @@ export const confirmPassword = async (
   return response;
 };
 
+type BalancesResponse = {
+  contractId?: string;
+  issuer?: {
+    key: string;
+  };
+  decimals: number;
+  name: string;
+  symbol: string;
+  valueXdr: string;
+}[];
+
+export const getAccountIndexerBalances = async (
```

Comment:
This takes the data from the indexer and formats it to match our current use from the RPCs, following the BalanceMap pattern.
I thought it would be best to do this here instead of in the backend because we only do this in order to be able to achieve the fallback strategy in a cleaner way and hopefully one day this can be removed. 
Open to suggestions on alternatives.